controllers/components/dashboard_controller.go

```diff
@@ -0,0 +1,460 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/opendatahub-io/opendatahub-operator/v2/apis/components"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	v1 "github.com/openshift/api/operator/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	dscv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/datasciencecluster/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	ComponentNameUpstream = "dashboard"
+	PathUpstream          = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/odh"
+
+	ComponentNameDownstream = "rhods-dashboard"
+	PathDownstream          = deploy.DefaultManifestPath + "/" + ComponentNameUpstream + "/rhoai"
+	PathSelfDownstream      = PathDownstream + "/onprem"
+	PathManagedDownstream   = PathDownstream + "/addon"
+	OverridePath            = ""
+	DefaultPath             = ""
+)
+
+var (
+	DSCIInstance *dsciv1.DSCInitialization
+	DSCInstance  *dscv1.DataScienceCluster
+)
+
+// DashboardReconciler reconciles a Dashboard object
+type DashboardReconciler struct {
+	*BaseReconciler
+	// Any dashboard specific reconcile request fields can be added below
+}
+
+func NewDashboardReconciler(mgr ctrl.Manager) *DashboardReconciler {
+	r := &DashboardReconciler{
+		BaseReconciler: NewBaseReconciler(mgr, "dashboard"),
+	}
+
+	// Add Dashboard-specific actions
+	r.AddAction(&InitializeAction{})
+	r.AddAction(&SupportDevFlagsAction{})
+	r.AddAction(&CleanupOAuthClientAction{})
+	r.AddAction(&DeployComponentAction{})
+	r.AddAction(&UpdateStatusAction{})
```

Comment:
any reason we want to create multiple
 type -> only one method Execute -> to implement interface Action ?

Reply:
@lburgazzoli Updated it in here https://github.com/VaishnaviHire/opendatahub-operator/pull/14/files#diff-bb31658616728a95856129f48054a0e7856e4b69ac6a9e3ee8f345e12ae582eaR75